Web3/Classes/Core/Transaction/EthereumTransaction.swift

```diff
@@ -9,7 +9,116 @@
 import Foundation
 import BigInt
 
-public struct EthereumTransaction {
+public struct EthereumTransaction: Codable {
+    /// The number of transactions made prior to this one
+    public var nonce: EthereumQuantity?
+    
+    /// Gas price provided Wei
+    public var gasPrice: EthereumQuantity?
+    
+    /// Gas limit provided
+    public var gasLimit: EthereumQuantity?
+    
+    /// Address of the sender
+    public var from: EthereumAddress?
+    
+    /// Address of the receiver
+    public var to: EthereumAddress?
+    
+    /// Value to transfer provided in Wei
+    public var value: EthereumQuantity?
+    
+    /// Input data for this transaction
+    public var data: EthereumData
+    
+    // MARK: - Initialization
+    
+    /**
+     * Initializes a new instance of `EthereumTransaction` with the given values.
+     *
+     * - parameter nonce: The nonce of this transaction.
+     * - parameter gasPrice: The gas price for this transaction in wei.
+     * - parameter gasLimit: The gas limit for this transaction.
+     * - parameter from: The address to send from, required to send a transaction using sendTransaction()
+     * - parameter to: The address of the receiver.
+     * - parameter value: The value to be sent by this transaction in wei.
+     * - parameter data: Input data for this transaction. Defaults to [].
+     */
+    init(
+        nonce: EthereumQuantity? = nil,
+        gasPrice: EthereumQuantity? = nil,
+        gasLimit: EthereumQuantity? = nil,
+        from: EthereumAddress? = nil,
+        to: EthereumAddress? = nil,
+        value: EthereumQuantity? = nil,
+        data: EthereumData = EthereumData(bytes: [])
+    ) {
+        self.nonce = nonce
+        self.gasPrice = gasPrice
+        self.gasLimit = gasLimit
+        self.from = from
+        self.to = to
+        self.value = value
+        self.data = data
+    }
+    
+    
+    // MARK: - Convenient functions
+    
+    /**
+     * Signs this transaction with the given private key and discards old signatures if present.
```

Comment:
Could you update this line please? It doesn't discard "old signatures" any more as EthereumTransaction doesn't contain a signature.

Reply:
Ah, yes definitely. I'll update this.